### ISSUE TITLE
Crash in SDL Router service class

### ISSUE THREAD

**Issue by ashwink11** [smartdevicelink/sdl_java_suite#1239]
### Bug Report
ClassCastException in SDL router service class

##### Reproduction Steps
Not known, received crashed report for the released app.

##### Expected Behavior
The app should not crash.

##### Observed Behavior
Crash issue reported.

##### OS & Version Information
* Android Version: OnePlus 5 with Android 9
* SDL Android Version: 4.10.0
* Testing Against: Not known, received crashed report for the released app.

##### Crash Logs

```
Fatal Exception: java.lang.ClassCastException: java.lang.Object cannot be cast to java.util.List
       at com.smartdevicelink.transport.SdlRouterService$RegisteredApp.getCompatPrimaryTransport + 3222(SdlRouterService.java:3222)
       at com.smartdevicelink.transport.SdlRouterService$RegisteredApp.handleIncommingClientMessage + 3248(SdlRouterService.java:3248)
       at com.smartdevicelink.transport.SdlRouterService$RouterHandler$1.run + 496(SdlRouterService.java:496)
       at java.util.concurrent.ThreadPoolExecutor.runWorker + 1167(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run + 641(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run + 764(Thread.java:764)
```

**Comment by timeyinxingren** [comment]
what's the released app,how to reproduce

**Comment by E-SAITO-TMC** [comment]
@joeygrover -san @theresalech -san This issue is a top priority for Toyota. This has little information, can it be reproduced? I want to know if the issue will be reproduced in the latest version.

**Comment by E-SAITO-TMC** [comment]
@joeygrover -san  We can't reproduce this issue due to unknown procedure,  Would you like to leave this bug open?

**Comment by theresalech** [comment]
@ashwink11 can you please advise if you are still seeing this issue?  If so, please provide reproduction steps.  If not, we will close.

**Comment by ashwink11** [comment]
Hi @theresalech , I am not able to reproduce this issue. This was one of the issues seen by users of the What3words app. The app developers received info by a crash reporting tool. Unfortunately, they removed SDL support due to issues, hence I cannot confirm if this issue is still reproducible.

**Comment by theresalech** [comment]
As we have been unable to reproduce this issue, we will close.  If the issue can be reproduced in the future, and reproduction steps can be provided, a new issue should be submitted.

**Comment by decubate** [comment]
We are seeing this crash in our crash reporting tool as well, but have as of yet not been able to reproduce it.

**Comment by joeygrover** [comment]
We are still getting reports that this bug exists. However, there are no reproduction steps to test it exists or even if a fix would solve the problem. I am going to reopen the issue and attempt to add safe guards around the code in question, but without those steps there will be no guarantee the issue is addressed.

**Comment by theresalech** [comment]
Closed by #1660.

**Comment by asheeshs** [comment]
We are also seeing this crash `Fatal Exception: java.lang.ClassCastException`

Stats: 
- SDL Version: 4.12.0
- Android versions: 40% Android 9, 34% Android 11, 18% Android 10
- App is 100% in background

```
Fatal Exception: java.lang.ClassCastException: java.lang.Object cannot be cast to java.util.List
       at com.smartdevicelink.transport.SdlRouterService$RegisteredApp.getCompatPrimaryTransport(SdlRouterService.java:3324)
       at com.smartdevicelink.transport.SdlRouterService$RegisteredApp.handleIncommingClientMessage(SdlRouterService.java:3350)
       at com.smartdevicelink.transport.SdlRouterService$RouterHandler$1.run(SdlRouterService.java:504)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:920)
```

Stats: 
- SDL Version: 4.12.0
- Android versions: 47% Android 10, 44% Android 11, 6% Android 9
- App is 100% in background

```
Fatal Exception: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
       at java.util.ArrayList.get(ArrayList.java:437)
       at com.smartdevicelink.transport.SdlRouterService$RegisteredApp.getCompatPrimaryTransport(SdlRouterService.java:3326)
       at com.smartdevicelink.transport.SdlRouterService$RegisteredApp.handleIncommingClientMessage(SdlRouterService.java:3350)
       at com.smartdevicelink.transport.SdlRouterService$RouterHandler$1.run(SdlRouterService.java:504)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)
```

Unfortunately I don't have reproduction steps, aboe stats are pulled from Crashlytics dash.

**Comment by joeygrover** [comment]
@asheeshs your comment states you are using 4.12.0 as the SDL Library, since the fix went out in 5.1.1 you will need to update or else you will continue to see these issues. Please update your library to the latest version as we have provided potential fixes to these issues. If you are still seeing these issues in your reports after you upgrade to the latest version then let us know and we will investigate.

**Comment by asheeshs** [comment]
Thanks @joeygrover, didn't notice its already fixed in 5.1.1. Please go ahead and close this issue.
